test/DGmethods/courant.jl

```diff
@@ -0,0 +1,170 @@
+#                end
+
+                Q = init_ode_state(dg, FT(0))
+                solver = LSRK54CarpenterKennedy(dg, Q; dt = Δt, t0 = 0)
+                solve!(Q, solver; timeend=Δt)
+                Δx = min_node_distance(grid, EveryDirection())
+                Δx_v = min_node_distance(grid, VerticalDirection())
+                Δx_h = min_node_distance(grid, HorizontalDirection())
+
+                T∞ = FT(300)
+                translation_speed = FT(212.13203435596427)
+                diff_speed_h = 734.8469228349534
+                diff_speed_v = 0
+                c   = Δt*(translation_speed + soundspeed_air(T∞))/Δx
+                c_h = Δt*(translation_speed + soundspeed_air(T∞))/Δx_h
+                c_v = Δt*(soundspeed_air(T∞))/Δx_v
+                d_h = Δt*diff_speed_h/Δx_h^2
+                d_v = Δt*diff_speed_v/Δx_v^2
+
+                if (FT==Float64 && dim == 2)
+                    @test abs(courant(nondiffusive_courant, dg, model, Q, Δt, HorizontalDirection()) - c_h) <= 1e-15
+                    @test abs(courant(diffusive_courant,    dg, model, Q, Δt, HorizontalDirection()) - d_h) <= 1e-13
+                    @test abs(courant(nondiffusive_courant, dg, model, Q, Δt, VerticalDirection()) - c_v) <= 1e-16
+                    @test abs(courant(diffusive_courant,    dg, model, Q, Δt, VerticalDirection()) - d_v) <= 1e-17
+                elseif (FT==Float64 && dim == 3)
+                    @test abs(courant(nondiffusive_courant, dg, model, Q, Δt, HorizontalDirection()) - c_h) <= 1e-11
+                    @test abs(courant(diffusive_courant,    dg, model, Q, Δt, HorizontalDirection()) - d_h) <= 1e-13
+                    @test abs(courant(nondiffusive_courant, dg, model, Q, Δt, VerticalDirection()) - c_v) <= 1e-11
+                    @test abs(courant(diffusive_courant,    dg, model, Q, Δt, VerticalDirection()) - d_v) <= 1e-17
+                elseif (dim == 2)
+                    @test abs(courant(nondiffusive_courant, dg, model, Q, Δt, HorizontalDirection()) - c_h) <= 1e-11
+                    @test abs(courant(diffusive_courant,    dg, model, Q, Δt, HorizontalDirection()) - d_h) <= 1e-9
+                    @test abs(courant(nondiffusive_courant, dg, model, Q, Δt, VerticalDirection()) - c_v) <= 1e-16
+                    @test abs(courant(diffusive_courant,    dg, model, Q, Δt, VerticalDirection()) - d_v) <= 1e-17
+                else
+                    @test abs(courant(nondiffusive_courant, dg, model, Q, Δt, HorizontalDirection()) - c_h) <= 1e-11
+                    @test abs(courant(diffusive_courant,    dg, model, Q, Δt, HorizontalDirection()) - d_h) <= 1e-7
+                    @test abs(courant(nondiffusive_courant, dg, model, Q, Δt, VerticalDirection()) - c_v) <= 1e-11
+                    @test abs(courant(diffusive_courant,    dg, model, Q, Δt, VerticalDirection()) - d_v) <= 2e-9
+                end
```

Comment:
No tests for `advective_courant`?

Reply:
I believe, for the purposes of estimating the time step size, only nondiffusive and diffusive courant functions are need. Looks like advective courant seems to be added only for diagnostics purposes.  
@slifer50 : Please correct me if I am missing anything. If the target is courant number preset, this can be safely removed.

Reply:
@tapios seems to suggest that both are needed [here](https://github.com/climate-machine/CLIMA/pull/618#issuecomment-573333724).

Reply:
The non diffusive courant number adds up a courant number from the sound waves and the advective courant number, if it works correctly it implies that the advective number should be correct aswell.

Reply:
@skandalaCLIMA When we have multirate time stepping, and if we want to make that adaptive, we'll need the advective Courant number for the dynamical time steps, and acoustic Courant number for fast steps.

Reply:
All three courant number functions are added per reviewer recommendation.